svgdigitizer/__main__.py

```diff
@@ -31,19 +31,46 @@ def cli(): pass
 @click.argument('svg', type=click.File('rb'))
 def plot(svg, sampling_interval):
     from svgdigitizer.svgplot import SVGPlot
-    SVGPlot(svg, sampling_interval=sampling_interval).plot()
+    from svgdigitizer.svg import SVG
+    SVGPlot(SVG(svg), sampling_interval=sampling_interval).plot()
 
 
 @click.command()
 @click.option('--sampling_interval', type=float, default=None, help='specify sampling interval (for now in mV)')
 @click.argument('svg', type=click.Path(exists=True))
 def digitize(svg, sampling_interval):
     from svgdigitizer.svgplot import SVGPlot
-    plot = SVGPlot(open(svg, 'rb'), sampling_interval=sampling_interval)
+    from svgdigitizer.svg import SVG
+    plot = SVGPlot(SVG(open(svg, 'rb')), sampling_interval=sampling_interval)
     from pathlib import Path
     plot.df.to_csv(Path(svg).with_suffix('.csv'), index=False)
 
 
+@click.command()
+@click.option('--sampling_interval', type=float, default=None, help='specify sampling interval (for now in mV)')
+@click.option('--metadatayaml', type=click.Path(exists=True), default=None, help='specify a yaml file with metadata')
+@click.argument('svg', type=click.Path(exists=True))
+def cv(svg, sampling_interval, metadatayaml):
+    import yaml
+    from svgdigitizer.svgplot import SVGPlot
+    from svgdigitizer.svg import SVG
+    from svgdigitizer.electrochemistry.cv import CV
+    if metadatayaml:
+        with open(metadatayaml) as f:
+            metadata = yaml.load(f, Loader=yaml.FullLoader)
+
+        cv = CV(SVGPlot(SVG(open(svg, 'rb')), sampling_interval=sampling_interval), metadata=metadata)
+
+    else:
+        cv = CV(SVGPlot(SVG(open(svg, 'rb')), sampling_interval=sampling_interval))
+
+    from pathlib import Path
+    cv.cv_df.to_csv(Path(svg).with_suffix('.csv'), index=False)
+    import json
+    with open(Path(svg).with_suffix('.json'), "w") as outfile:
+        json.dump(cv.metadata_out, outfile)
```

Comment:
```suggestion
@click.option('--metadata', type=click.File("rb"), default=None, help='yaml file with metadata')
@click.argument('svg', type=click.Path(exists=True))
def cv(svg, sampling_interval, metadata):
    import yaml
    from svgdigitizer.svgplot import SVGPlot
    from svgdigitizer.svg import SVG
    from svgdigitizer.electrochemistry.cv import CV
    if metadata:
        metadata = yaml.load(metadata, Loader=yaml.FullLoader)

    cv = CV(SVGPlot(SVG(open(svg, 'rb')), sampling_interval=sampling_interval), metadata=metadata)

    from pathlib import Path
    cv.cv_df.to_csv(Path(svg).with_suffix('.csv'), index=False)

    import json
    with open(Path(svg).with_suffix('.json'), "w") as outfile:
        json.dump(cv.metadata_out, outfile)
```

I think this can be simplified like this.

Reply:
I was not sure what `click.type(..)` would do. :)
The rest is fine.